cores/esp32/esp32-hal-uart.c

```diff
@@ -66,27 +66,41 @@ static uart_t _uart_bus_array[3] = {
 };
 #endif
 
+static void _uart_rx_flush( uart_t * uart){
```

Comment:
is "flush" the best name here? you are moving the data to the queue, not actually flushing it. Maybe `_uart_rx_read_fifo` is better?